k256/Cargo.toml

```diff
@@ -23,13 +23,22 @@ optional = true
 default-features = false
 
 [dev-dependencies]
-fiat-crypto = "0.1.0"
 hex = "0.4"
 proptest = "0.10"
+num-bigint = "0.3"
+num-traits = "0.2"
 
 [features]
 default = ["arithmetic", "std"]
+all-nonconflicting = ["default", "rand", "test-vectors", "64bit"]
 arithmetic = []
+field-10x26 = []
+field-5x52 = []
+field-montgomery = []
+scalar-8x32 = []
+scalar-4x64 = []
+64bit = ["field-5x52", "scalar-4x64"]
+32bit = ["field-10x26", "scalar-8x32"]
```

Comment:
Would it make sense to select the scalar/field backends automatically based on `cfg(target_pointer_width = ...)` instead of using Cargo features? As you're encountering here, they're really intended for additive features and not ones that are mutually exclusive.

Here's an example of the `polyval` crate doing this for its GF(2^128) field backends:

https://github.com/RustCrypto/universal-hashes/blob/master/polyval/src/field.rs#L46-L50

(still kind of gross, but at least automatic based on the target)

Reply:
Ok, I'll try it out. There's still a problem of how to fit the old (Montgomery) field arithmetic. Without it, we could just make 64-bit backends default, and 32-bit ones features that would turn off the 64-bit ones (that's the way I had it initially).

Reply:
The `polyval` example above is trying to juggle three field arithmetic backends as well (32-bit, 64-bit, and one based on Intel AVX2). To make that a bit more manageable it's using the  [`cfg-if`](https://docs.rs/cfg-if/) crate to manage some of the gating.

It seems like you'll need at least one Cargo feature in the mix to keep the Montgomery field arithmetic around, but unless there's a good reason to do otherwise it seems like you can autodetect the 32/64-bit distinction based on the target.

Reply:
Hm, in this case how would I go around running tests for 32 bit target on my machine? I'm on Macos, and I don't see any mac-32 bit target (Catalina doesn't even support 32-bit anymore afaik). Would it make sense to add a "force 32 bit" feature or something?

Reply:
And how would the CI config look like?

Reply:
Ok, made another attempt, with a `force-32-bit` flag to test 32-bit backends.

Reply:
The absence of a 32-bit target for macOS is a bit annoying (I sympathize as a fellow macOS user). Otherwise target-based gating could work for CI (and checking on Linux for local development).

Another option is to add `cfg(test)` to the gating for including the backends, and include both under a `test` configuration.

Since this has gotten a lot trickier than I initially thought it was going to be, I'm happy to table it for now with your `force-32-bit` feature and circle back on it myself and see if I can wire up things using `cfg-if` to test both the 32-bit and 64-bit backends, but select one of them to use automatically in builds.

Reply:
Ok, let's call it fixed for now.